crates/teloxide/src/lib.rs

```diff
@@ -30,7 +30,56 @@
 //!     <img src=https://github.com/teloxide/teloxide/raw/master/media/throw-dice.gif width=420px />
 //!   </kbd>
 //! </div>
+//! 
+//! ## Working with Updates and Messages
+//! There're a great number of [update kinds](https://docs.rs/teloxide/latest/teloxide/types/enum.UpdateKind.html) and 
+//! [message kinds](https://docs.rs/teloxide/latest/teloxide/types/enum.MessageKind.html) to work with! Usually it's essential to filter specific ones
+//! and process them in *handler functions*. *Teloxide* provides some `filter methods` for `Update` and `Message` types in [UpdateFilterExt](https://docs.rs/teloxide/latest/teloxide/dispatching/trait.UpdateFilterExt.html) 
+//! and [MessageFilterExt](https://docs.rs/teloxide/latest/teloxide/dispatching/trait.MessageFilterExt.html) traits respectively. In addition to filtering, these
+//! methods will inject the appropriate type into your handler functions. For instance, if you use `Update::filter_message`, the `Message` object will be available as a parameter
+//! at your handler functions. Analogously the use of `Message::filter_text` will inject the `String` object.
+//! 
+//! (Note: `filter_text` actually uses a function that returns Option<&str> value, so every filter_.. fn *always* return an `Owned` version of a  type)
```

Comment:
I am not sure why this is needed here?
```suggestion
```

Reply:
Initially I didn't get the actual purpose of filter_ fns other than filtering out updates according to a condition. So, when I saw that I can use it to inject values, it was misleading: why not just using borrowed values.

I think it should be mentioned that the value a handler can access is an owned one. Maybe it worth be introduced as a comment